services/121-service/src/transaction-job-processors/transaction-job-processors.service.ts

```diff
@@ -217,30 +215,151 @@ export class TransactionJobProcessorsService {
       programId: input.programId,
       paymentNumber: input.paymentNumber,
       userId: input.userId,
-      calculatedTranserAmountInMajorUnit:
+      calculatedTransferAmountInMajorUnit:
         intersolveVisaDoTransferOrIssueCardReturnDto.amountTransferredInMajorUnit,
       financialServiceProviderId: financialServiceProvider.id,
       registration,
       oldRegistration,
       isRetry: input.isRetry,
-      status: StatusEnum.success,
+      status: TransactionStatusEnum.success,
     });
   }
 
+  public async processSafaricomTransactionJob(
```

Comment:
Do you think I should check the logic of this function against the Activity Diagram, to confirm from my side as well conceptually we have it covered?

Reply:
I feel this has been sufficiently checked, but up to you.